extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java

```diff
@@ -274,7 +288,7 @@ void ignoreWarnings(BuildProducer<ReflectiveHierarchyIgnoreWarningBuildItem> ign
     void registerClasses(
             BuildProducer<NativeImageSystemPropertyBuildItem> nativeImageProps,
             BuildProducer<ServiceProviderBuildItem> providerItem,
-            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveClass,
```

Comment:
My guess is that this change should be reverted and that's probably what causes the warnings.